packages/core/core/src/BundleGraph.js

```diff
@@ -489,15 +498,17 @@ export default class BundleGraph {
 
   findReachableBundleWithAsset(bundle: Bundle, asset: Asset) {
     let bundleGroups = this.getBundleGroupsContainingBundle(bundle);
-
     for (let bundleGroup of bundleGroups) {
       // If the asset is in any sibling bundles, return that bundle.
-      let bundles = this.getBundlesInBundleGroup(bundleGroup);
-      let res = bundles.find(
-        b => b.id !== bundle.id && this.bundleHasAsset(b, asset),
-      );
-      if (res != null) {
-        return res;
+      let bundles = this.getBundlesInBundleGroup(bundleGroup).reverse();
```

Comment:
The remaining question: why is `reverse()` needed here but for example not when inserting the sibling `<script>` tags (where the order should be wrong as well then).